app/src/main/java/fly/speedmeter/grub/Data.java

```diff
@@ -50,49 +52,79 @@ public Data(onGpsServiceUpdate onGpsServiceUpdate){
     public void addDistance(double distance){
         distanceM = distanceM + distance;
         distanceKm = distanceM / 1000f;
+        distanceMiles = distanceM / 1609.34;
     }
 
-    public SpannableString getDistance(){
+    public SpannableString getDistance(boolean isMPH){
         SpannableString s;
-        if (distanceKm < 1) {
-            s = new SpannableString(String.format("%.0f", distanceM) + "m");
-            s.setSpan(new RelativeSizeSpan(0.5f), s.length() - 1, s.length(), 0);
-        }else{
+        if(isMPH && distanceMiles >= 1){
```

Comment:
I suggest you to do a first if with only the isMPH and then a nested one with the condition related to it. Your current implementation can lead to errors.